File: HELLO_AJAX/HELLO_AJAX/daoemp.py
import pymysql

class DaoEmp:
    def __init__(self):
        self.conn = pymysql.connect(host='localhost', user='root', password='python',
                       port=3305, db='python', charset='utf8')
        self.curs = self.conn.cursor(pymysql.cursors.DictCursor)
        
    def selectList(self):
        sql = "select * from emp"
        self.curs.execute(sql)
         
        list = self.curs.fetchall()
        return list
    
    def selectOne(self, e_id):
        sql = f"""
                select * from emp 
                where 
                    e_id = '{e_id}'
        """
        self.curs.execute(sql)
        vo = self.curs.fetchone()
        return vo
    
    
    def insert(self, e_id, e_name, gen, addr):
        sql = f"""
                insert into emp
                (e_id, e_name, gen, addr)
                values 
                ('{e_id}', '{e_name}', '{gen}', '{addr}')
        """
        cnt = self.curs.execute(sql)
        self.conn.commit()
        return cnt
    
    
    def update(self, e_id, e_name, gen, addr):
        sql = f"""
                update emp
                set
                    e_name = '{e_name}', 
                    gen = '{gen}',
                    addr = '{addr}'
                where e_id = '{e_id}'
        """
        cnt = self.curs.execute(sql)
        self.conn.commit()
        return cnt
    
    
    def delete(self, e_id):
        sql = f"""
                delete from emp
                where e_id = '{e_id}'
        """
        cnt = self.curs.execute(sql)
        self.conn.commit()
        return cnt
         
        
    def __del__(self):
        self.curs.close()
        self.conn.close()

if __name__ == '__main__':
    de = DaoEmp()
    
    list = de.selectList()
    print(list)
    
    vo = de.selectOne('1')
    print(vo)
    
    cnt = de.insert('5','5','5','5')
    print("insert cnt", cnt)
    
    cnt = de.update('5','6','6','6')
    print("update cnt", cnt)
     
    cnt = de.delete('5')
    print("delete cnt", cnt)
    
    
    

File: HELLO_EXPRESS_MVVM/daoemp.js
var Mysql = require('sync-mysql');

class DaoEmp {
	
	constructor() {
		this.conn = new Mysql({
			host: 'localhost',
			port: 3305,
			user: 'root',
			password: 'python',
			database: 'python'
		});

	}
	
	selectList() {
		var sql = `
			select * from emp
		`;
		var result = this.conn.query(sql);
		return result;
	}
	
	select(e_id) {
		var sql = `
			select * 
				from emp
				where e_id = '${e_id}'
		`;
		var result = this.conn.query(sql);
		return result[0];
	}
	
	insert(e_id, e_name, gen, addr) {
		var sql = `
			insert into emp
				(e_id, e_name, gen, addr)
			values
				('${e_id}', '${e_name}', '${gen}', '${addr}')
		`;
		var result = this.conn.query(sql);
		return result.affectedRows;
	}
	
	update(e_id, e_name, gen, addr) {
		var sql = `
			update emp
				set e_name='${e_name}', 
					gen='${gen}',
					addr='${addr}'
				where e_id='${e_id}'
		`;
		var result = this.conn.query(sql);
		return result.affectedRows;
	}
	
	delete(e_id) {
		var sql = `
			delete from emp
				where e_id='${e_id}'
		`;
		var result = this.conn.query(sql);
		return result.affectedRows;
	}
}

module.exports = DaoEmp;

if (require.main === module) {
    var de = new DaoEmp();
	var list = de.selectList();
	var vo = de.select('1');
	var insertCnt = de.insert('3','3','3','3');
	var updateCnt = de.update('3','4','4','4');
	var deleteCnt = de.delete('3');
	console.log(list);
	console.log(vo);
	console.log(insertCnt);
	console.log(updateCnt);
	console.log(deleteCnt);
	
}
